controllers/accountclaim/accountclaim_controller.go

```diff
@@ -821,57 +822,95 @@ func (r *AccountClaimReconciler) getUnclaimedAccount(reqLogger logr.Logger, acco
 		reqLogger.Info(fmt.Sprintf("defaultAccountPoolName: %s", defaultAccountPoolName))
 	}
 
-	if accountClaim.Spec.AccountPool == defaultAccountPoolName || accountClaim.Spec.AccountPool == "" {
-		for _, account := range accountList.Items {
-			// Ensure we're pulling accounts from the default accountPool
-			if account.Spec.AccountPool == defaultAccountPoolName || (account.IsOwnedByAccountPool() && account.Spec.AccountPool == "") {
-				acc, _ := CheckClaimAccountValidity(reqLogger, account, accountClaim)
-				if acc != nil {
-					return acc, nil
-				}
-			}
-		}
-	} else {
-		for _, account := range accountList.Items {
-			if account.Spec.AccountPool == accountClaim.Spec.AccountPool {
-				acc, _ := CheckClaimAccountValidity(reqLogger, account, accountClaim)
-				if acc != nil {
-					return acc, nil
-				}
-			}
-		}
-	}
+  var unusedAccount *awsv1alpha1.Account
+
+  for _, loopAccount := range accountList.Items {
+    // assign to new variable to prevent issues with using a pointer to the loop var later
+    account := loopAccount
+    if ! IsSameAccountPoolNames(account.Spec.AccountPool, accountClaim.Spec.AccountPool, defaultAccountPoolName) {
+      continue
+    }
+
+    if ! CanAccountBeClaimedByAccountClaim(&account, accountClaim) {
+      continue
+    }
 
+    if account.Status.Reused {
+			reqLogger.Info(fmt.Sprintf("Reusing account: %s", account.ObjectMeta.Name))
```

Comment:
Is the indentation here (and 847) just a weird formatting thing?

Reply:
I ran go fmt again for the files and now it should be good.